docs(linter): Add config docs for no-restricted-globals rule.

connorshea opened the pull request:
Part of #14743.

Generated docs:

```md

## Configuration

This rule accepts a configuration object with the following properties:

### restrictedGlobals

type: `Record<string, string>`

default: `{}`

Objects in the format
`{ "name": "event", "message": "Use local parameter instead." }`, which define what globals
are restricted from use.
```

This is technically a positional argument and so should maybe be held back until we have a solution for documenting this kind of configuration option?

camc314 commented:
### Merge activity

* **Nov 13, 9:04 AM UTC**: The merge label '0-merge' was detected. This PR will be added to the [Graphite merge queue](https://app.graphite.com/merges?org=oxc-project&repo=oxc) once it meets the requirements.
* **Nov 13, 9:04 AM UTC**: `camc314` added this pull request to the [Graphite merge queue](https://app.graphite.com/merges?org=oxc-project&repo=oxc).
* **Nov 13, 9:09 AM UTC**: Merged by the [Graphite merge queue](https://app.graphite.com/merges?org=oxc-project&repo=oxc).

connorshea committed:
docs(linter): Add config docs for no-restricted-globals rule. (#15662)

Part of #14743.

Generated docs:

```md

## Configuration

This rule accepts a configuration object with the following properties:

### restrictedGlobals

type: `Record<string, string>`

default: `{}`

Objects in the format
`{ "name": "event", "message": "Use local parameter instead." }`, which define what globals
are restricted from use.
```

This is technically a positional argument and so should maybe be held back until we have a solution for documenting this kind of configuration option?
